tests/acceptance/stepDefinitions/sharingContext.js

```diff
@@ -283,6 +290,88 @@ Then('all users and groups that contain the string {string} in their name should
     })
 })
 
+Then('all users and groups that contain the string {string} in their name should be listed in the autocomplete list on the webUI except user {string}', function (pattern, alreadySharedUser) {
+  return client.page.FilesPageElement.sharingDialog().getShareAutocompleteItemsList()
+    .then(itemsList => {
+      const currentUserDisplayName = userSettings.getDisplayNameForUser(client.globals.currentUser)
+      // check if all created users that contain the pattern either in the display name or the username
+      // are listed in the autocomplete list
+      // in both cases the display name should be listed
+      for (var userId in userSettings.getCreatedUsers()) {
+        const userDisplayName = userSettings.getDisplayNameForUser(userId)
+        if ((userDisplayName.toLowerCase().includes(pattern) ||
+          userId.includes(pattern)) &&
+          (userDisplayName !== currentUserDisplayName) &&
+        userDisplayName !== alreadySharedUser) {
+          const userString = userDisplayName + client.page.FilesPageElement.sharingDialog().getUserSharePostfix()
+
+          assert.ok(
+            itemsList.includes(userString),
+            `could not find '${userDisplayName}' in autocomplete share list`
+          )
+        }
+      }
+      assert.ok(!itemsList.includes(alreadySharedUser), `"${alreadySharedUser}" was listed in the autocompletion list but should not have been`)
+
+      // check if every created group that contains the pattern is listed in the autocomplete list
+      userSettings.getCreatedGroups().forEach(function (groupId) {
+        if (groupId.toLowerCase().includes(pattern)) {
+          const groupString = groupId + client.page.FilesPageElement.sharingDialog().getGroupSharePostfix()
+
+          assert.ok(
+            itemsList.includes(groupString),
+            `could not find '${groupString}' in autocomplete share list`
+          )
+        }
+      })
+    })
+})
+
+Then('all users and groups that contain the string {string} in their name should be listed in the autocomplete list on the webUI except group {string}', function (pattern, alreadySharedGroup) {
```

Comment:
This function looks exactly the same like the above function, find a way to minimize the duplicate codes